pkg/agent/controller/activeprobe/packetin.go

```diff
@@ -0,0 +1,169 @@
+	ActiveProbeRegRigthIndex uint32 = 8
+)
+
+func (a *Controller) HandlePacketIn(packetIn *ofctrl.PacketIn) error {
+	ap := activeprobev1alph1.ActiveProbe{}
+	reason := ""
+
+	state, tag, apResult, err := a.parsePacketIn(packetIn)
+	if err != nil {
+		klog.Errorf("parsePacketIn error: %+v", err)
+		return err
+	}
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		name := a.RunningActiveprobe[tag].name
+		namespacedName := types.NamespacedName{
+			Namespace: "",
+			Name:      name,
+		}
+		if err := a.K8sClient.Get(context.TODO(), namespacedName, &ap); err != nil {
+			klog.Warningf("Update ActiveProbe failed: %+v", err)
+		}
+		ap.Status.State = state
+		ap.Status.Tag = tag
+
+		err = a.updateActiveProbeStatus(&ap, apResult, reason)
+		if err != nil {
+			klog.Warningf("Update ActiveProbe failed: %+v", err)
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		klog.Errorf("Update ActiveProbe error: %+v", err)
+	}
+
+	return err
+}
+
+func (a *Controller) parsePacketIn(packetIn *ofctrl.PacketIn) (activeprobev1alph1.ActiveProbeState, uint8, *activeprobev1alph1.AgentProbeResult, error) {
+	var err error
+	var tag uint8
+	var reg4Val, reg5Val uint32
+	var inPort uint32
+	state := activeprobev1alph1.ActiveProbeFailed
+	var telemetryTracePoint activeprobev1alph1.TelemetryTracePoint
+	var activeProbeAction activeprobev1alph1.ActiveProbeAction
+	agentProbeResult := activeprobev1alph1.AgentProbeResult{}
+	activeProbeTracePoint := activeprobev1alph1.ActiveProbeTracePoint{}
+	matchers := packetIn.GetMatches()
+	if packetIn.Data.Ethertype == protocol.IPv4_MSG {
+		ipPacket, ok := packetIn.Data.Data.(*protocol.IPv4)
+		if !ok {
+			state = activeprobev1alph1.ActiveProbeFailed
+			return state, 0, nil, errors.New("invalid IPv4 packet")
+		}
+		tag = ipPacket.DSCP
+
+		if match := getMatchInPortField(matchers); match != nil {
+			inPort, err = getInportVal(match)
+			if err != nil {
+				return state, tag, nil, err
+			}
+		}
+
+		flowActionField := ofctrl.NewRegField(FlowActionRegID, FlowActionRegLeftIndex, FlowActionRegRigthInnex, FlowAction)
+		activeProbeField := ofctrl.NewRegField(ActiveProbeRegID, ActiveProbeRegLeftIndex, ActiveProbeRegRigthIndex, ActiveProbe)
+
+		reg4Val, err = getRegValue(matchers, flowActionField)
+		if err != nil {
+			return state, tag, nil, err
+		}
+		if reg4Val == datapath.PolicyRuleActionAllow {
+			activeProbeAction = activeprobev1alph1.ActiveProbeAllow
+		} else if reg4Val == datapath.PolicyRuleActionDeny {
+			activeProbeAction = activeprobev1alph1.ActiveProbeDrop
+		}
+
+		reg5Val, err = getRegValue(matchers, activeProbeField)
+		if err != nil {
+			return state, tag, nil, err
+		}
+
+		switch {
+		case reg5Val == datapath.Tier1PolicyMatch && inPort == datapath.POLICY_TO_LOCAL_PORT:
+			telemetryTracePoint = activeprobev1alph1.IsolationPolicyEgress
+		case reg5Val == datapath.Tier2PolicyMatch && inPort == datapath.POLICY_TO_LOCAL_PORT:
+			telemetryTracePoint = activeprobev1alph1.FroensicPolicyEgress
```

Comment:
FroensicPolicyEgress -> ForensicPolicyEgress
and in activeprobe/v1alpha1 types.go FroensicPolicyEgresss -> ForenscPolicyEgress

Reply:
has been modified